tools/src/test/java/org/apache/kafka/tools/consumer/group/DescribeConsumerGroupTest.java

```diff
@@ -16,88 +16,100 @@
-    private static final List<List<String>> DESCRIBE_TYPES;
-
-    static {
-        List<List<String>> describeTypes = new ArrayList<>();
+    private static final List<List<String>> DESCRIBE_TYPES = Stream.of(DESCRIBE_TYPE_OFFSETS, DESCRIBE_TYPE_MEMBERS, DESCRIBE_TYPE_STATE).flatMap(Collection::stream).collect(Collectors.toList());
+    private final ClusterInstance clusterInstance;
 
-        describeTypes.addAll(DESCRIBE_TYPE_OFFSETS);
-        describeTypes.addAll(DESCRIBE_TYPE_MEMBERS);
-        describeTypes.addAll(DESCRIBE_TYPE_STATE);
+    DescribeConsumerGroupTest(ClusterInstance clusterInstance) {
+        this.clusterInstance = clusterInstance;
+    }
 
-        DESCRIBE_TYPES = describeTypes;
+    private static List<ClusterConfig> generator() {
+        return ConsumerGroupCommandTestUtils.generator();
     }
 
-    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
-    @MethodSource({"getTestQuorumAndGroupProtocolParametersAll"})
-    public void testDescribeNonExistingGroup(String quorum, String groupProtocol) {
-        createOffsetsTopic(listenerName(), new Properties());
+    @ClusterTemplate("generator")
+    public void testDescribeNonExistingGroup() {
         String missingGroup = "missing.group";
 
         for (List<String> describeType : DESCRIBE_TYPES) {
             // note the group to be queried is a different (non-existing) group
-            List<String> cgcArgs = new ArrayList<>(Arrays.asList("--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--group", missingGroup));
+            List<String> cgcArgs = new ArrayList<>(Arrays.asList("--bootstrap-server", clusterInstance.bootstrapServers(), "--describe", "--group", missingGroup));
             cgcArgs.addAll(describeType);
-            ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs.toArray(new String[0]));
-
-            String output = ToolsTestUtils.grabConsoleOutput(describeGroups(service));
-            assertTrue(output.contains("Consumer group '" + missingGroup + "' does not exist."),
-                "Expected error was not detected for describe option '" + String.join(" ", describeType) + "'");
+            try (ConsumerGroupCommand.ConsumerGroupService service = consumerGroupService(cgcArgs.toArray(new String[0]));) {
+                String output = ToolsTestUtils.grabConsoleOutput(describeGroups(service));
+                assertTrue(output.contains("Consumer group '" + missingGroup + "' does not exist."),
+                        "Expected error was not detected for describe option '" + String.join(" ", describeType) + "'");
+            }
         }
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testDescribeWithMultipleSubActions(String quorum) {
+    @ClusterTemplate("generator")
+    public void testDescribeWithMultipleSubActions() {
```

Comment:
Those tests are used to verify the input arguments, hence they don't need a true kafka service. Could you remove `ClusterTemplate` and add fake `--bootstrap-server` to speedup this test?

Reply:
Move `testDescribeWithMultipleSubActions`, `testDescribeWithStateValue`, and `testPrintVersion` to `DescribeConsumerGroupUnitTest`.